`rustracer` fails to build on unstable

**Describe the bug**
The rustracer package does not build on unstable.

**To Reproduce**
Steps to reproduce the behavior:
1. Try to install rust-racer from unstable; for example, `nix-env -i rustracer` when unstable is set as the default channel

**Expected behavior**
A substitute pulled from the build server, or successfully built locally if no substitute is available

**Screenshots**
Full output is attached. Relevant errors:

```
error[E0557]: feature has been removed
  --> /build/racer-2.1.40-vendor.tar.gz/rustc-ap-rustc_data_structures/src/lib.rs:18:12
   |
18 | #![feature(optin_builtin_traits)]
   |            ^^^^^^^^^^^^^^^^^^^^ feature has been removed
   |
   = note: renamed to `auto_traits`

error[E0658]: auto traits are experimental and possibly buggy
  --> /build/racer-2.1.40-vendor.tar.gz/rustc-ap-rustc_data_structures/src/sync.rs:30:9
   |
30 |         pub auto trait Send {}
   |         ^^^^^^^^^^^^^^^^^^^^^^
   |
   = note: see issue #13231 <https://github.com/rust-lang/rust/issues/13231> for more information
   = help: add `#![feature(auto_traits)]` to the crate attributes to enable

error[E0658]: auto traits are experimental and possibly buggy
  --> /build/racer-2.1.40-vendor.tar.gz/rustc-ap-rustc_data_structures/src/sync.rs:31:9
   |
31 |         pub auto trait Sync {}
   |         ^^^^^^^^^^^^^^^^^^^^^^
   |
   = note: see issue #13231 <https://github.com/rust-lang/rust/issues/13231> for more information
   = help: add `#![feature(auto_traits)]` to the crate attributes to enable

error: aborting due to 3 previous errors
```

**Additional context**
In my setup, the nix package manager is running on an Arch installation.

**Notify maintainers**
@jagajaga @ma27

**Metadata**
 - system: `"x86_64-linux"`
 - host os: `Linux 5.11.11-arch1-1, Arch Linux, noversion`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.3.10`
 - channels(root): `"nixpkgs-21.05pre281594.6fc2b7ecc2a"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixpkgs`

Maintainer information:
```yaml
# a list of nixpkgs attributes affected by the problem
attribute: rustracer
[rustracer-build-attempt.txt](https://github.com/NixOS/nixpkgs/files/6292103/rustracer-build-attempt.txt)

# a list of nixos modules affected by the problem
module:
```

--- a/pkgs/development/tools/rust/racer/default.nix
+++ b/pkgs/development/tools/rust/racer/default.nix
@@ -2,16 +2,16 @@
 
 rustPlatform.buildRustPackage rec {
   pname = "racer";
-  version = "2.1.40";
+  version = "2.1.44";
 
   src = fetchFromGitHub {
     owner = "racer-rust";
     repo = "racer";
     rev = "v${version}";
-    sha256 = "sha256-8Is+RBfcXKbGSFzYoolLHs30rxlNI//xVGEOhxP2TV8=";
+    sha256 = "sha256-EmxJg2QDpGZ5TbMy9y6P11LdMucBdvewkRewuUzccGM=";
   };
 
-  cargoSha256 = "sha256-iUomr9viCdZk4nV75/OP8vHtJpMbmy+pq1IbaA2lLmE=";
+  cargoSha256 = "sha256-kKQnpEashpIwrXubuZIpU+tzxFaUjr6jaVunYPqaHnM=";
 
   nativeBuildInputs = [ makeWrapper ];
   buildInputs = lib.optional stdenv.isDarwin Security;

